onnx/defs/tensor/defs.cc

```diff
@@ -1657,4 +1567,119 @@ ONNX_OPERATOR_SET_SCHEMA(
           updateOutputElemType(ctx, 0, TensorProto::INT64);
         }));
 
+static const char* Pad_ver10_doc = R"DOC(
+Given `data` tensor, pads, mode, and value.
+Example:
+  Insert 0 pads to the beginning of the second dimension.
+  data = [
+      [1.0, 1.2],
+      [2.3, 3.4],
+      [4.5, 5.7],
+  ]
+  pads = [0, 2, 0, 0]
+  output = [
+      [
+          [0.0, 0.0, 1.0, 1.2],
+          [0.0, 0.0, 2.3, 3.4],
+          [0.0, 0.0, 4.5, 5.7],
+      ],
+  ]
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    Pad,
+    10,
+    OpSchema()
+        .Attr(
+            "mode",
+            "Three modes: constant(default), reflect, edge",
+            AttributeProto::STRING,
+            std::string("constant"))
+        .SetDoc(Pad_ver10_doc)
+        .Input(0, "data", "Input tensor.", "T")
+        .Input(
+            1,
+            "pads",
+            "Tensor of integers indicating the number of padding elements to add or remove (if negative) "
+            "at the beginning and end of each axis. For 2D input tensor, it is the number of pixels. "
+            "`pads` should be a 1D tensor of shape [2 * input_rank] or a 2D tensor of shape [1, 2 * input_rank]. "
+            "`pads` format (1D example) should be as follow [x1_begin, x2_begin,...,x1_end, x2_end,...], "
+            "where xi_begin is the number of pixels added at the beginning of axis `i` and "
+            "xi_end, the number of pixels added at the end of axis `i`.",
+            "T1")
+        .Input(
+            2,
+            "value",
+            "(Optional) Rank 1 tensor containing 1 float indicating the value to be filled if the mode chosen is `constant` (by default it is 0.0f).",
```

Comment:
More clarification about this 1-D tensor please (though the previous one is also not clear enough :)). Its shape should be the same as "pads".

Reply:
@linkerzhang - Actually the shape need not be the same as 'pads' , this the equivalent of the previous 'value' attribute - it is just a single float value (a scalar essentially) - so it's shape should be either [] or [1] - I ll update it

Reply:
I thought you would like to pad different values in each direction. if it's a [] or [1], then let's clarify it as a "Scalar".

Reply:
It is explicitly mentioned as a "scalar" in the latest version